crates/ra_hir/src/code_model.rs

```diff
@@ -98,6 +99,23 @@ impl Crate {
         db.crate_graph()[self.id].display_name.as_ref().cloned()
     }
 
+    pub fn query_external_importables(
+        self,
+        db: &dyn DefDatabase,
+        query: &str,
+    ) -> impl Iterator<Item = Either<ModuleDef, MacroDef>> {
```

Comment:
Note: we might or might not want to refactor `ModuleDef` to to contain `MacroDef` (but keeeping ModuleDefId macro-free).

Reply:
Or maybe even go further and remove distinction between `ScopeDef` and `ModuleDef` altogether? I feel for IDE cases it maybe makes sense to not do fine-grained distinction here? We already have `Definition` in `ra_ide_db` for that purpose, but I am wondering if we should push it down to the hir layer?